### ISSUE TITLE
CPU usage toooo high

### ISSUE THREAD

**Issue by jinyu121** [eriklindernoren/PyTorch-YOLOv3#63]
Hi~ Thank you for your fantastic code!

But~ The CPU usage is too high. Well, for example:

![2018-09-08 15-52-25](https://user-images.githubusercontent.com/7346324/45252554-c3678400-b38a-11e8-8e9e-6ee0ede78659.png)

In my code, I solve this phenomenon to some extent. Please see [this commit in my code](https://github.com/jinyu121/PyTorchYOLOv2/commit/b1b53c08a32bc2f22d467ba689a5aeafc579f476). It is very easy. 
Basically, move `mask`, `conf_mask`, `tX`, and `gt_box` to GPU, and use `pred_boxes.detach()` rather than `pred_boxes.cpu().data`.

After that, CPU cools down~

![2018-09-08 17-19-22](https://user-images.githubusercontent.com/7346324/45252580-67512f80-b38b-11e8-9073-dbac3f83880c.png)

**Comment by lianuo** [comment]
great，how about the training result？can you share a predict result of a image？

**Comment by jinyu121** [comment]
@lianuo My code? Not fully debug yet. So.....

**Comment by ydixon** [comment]
I think it was intended. Need more profiling to see which method is faster.

**Comment by nnzy11** [comment]
> Hi~ Thank you for your fantastic code!
> 
> But~ The CPU usage is too high. Well, for example:
> 
> ![2018-09-08 15-52-25](https://user-images.githubusercontent.com/7346324/45252554-c3678400-b38a-11e8-8e9e-6ee0ede78659.png)
> 
> In my code, I solve this phenomenon to some extent. Please see [this commit in my code](https://github.com/jinyu121/PyTorchYOLOv2/commit/b1b53c08a32bc2f22d467ba689a5aeafc579f476). It is very easy.
> Basically, move , , , and to GPU, and use rather than .`mask``conf_mask``tX``gt_box``pred_boxes.detach()``pred_boxes.cpu().data`
> 
> After that, CPU cools down~
> 
> ![2018-09-08 17-19-22](https://user-images.githubusercontent.com/7346324/45252580-67512f80-b38b-11e8-9073-dbac3f83880c.png)

Hello, I have encountered the same problem, but your link has been invalid, would you like to send it again, or tell me how to modify it? If you can get your reply, I will be very happy. Thank you for your help!
